### DISCUSSION TITLE
How many feeds?

### DISCUSSION THREAD

**Discussion by RUHavingFun** [mrlt8/docker-wyze-bridge#47] [General]
I am running a Raspberry Pi 4 4GB with two camera feeds.  That seems to be the sweet spot for me.  They are HD and nice and stable.  

I am curious if anyone is running on more powerful setups with upwards of 10 cameras?  I tried to add a bunch originally and pulled back.  First I was seeing some MAX error (what is the MAX?) and then I realized the Pi just could not handle too much more.

**Comment by mrlt8** [comment]
I've run 5 cams on a Pi 4 without any issues and about 125% CPU usage.
Some other user said they could run two cams on a Pi 2, so I think the Pi 4 should be able to handle a few more.

Could you open a new issue with the max error? I don't think I've seen that before.

**Reply by RUHavingFun** [reply]
I have recently tried to reproduce the error and can not.  It must be from some config change that is not related to adding more cams.  I overloaded with 10 cams and they all start.  Buffer like crazy, but they start fine.

**Reply by RUHavingFun** [reply]
Also, two still seems like my limit without buffering.  Are you running with more memory or overclocking?

**Reply by RUHavingFun** [reply]
I am now overclocking and seem stable on 3.

over_voltage=6
arm_freq=2000
gpu_freq=750

**Reply by mrlt8** [reply]
No overclocking or extra ram on mine. Running Ubuntu 21.04 (aarch64).
Also have a couple of other services running in the background.

Have you tried running the container on a different system to see if the pi is the bottleneck?
`docker stats` should show cpu and memory usage.

**Reply by RUHavingFun** [reply]
I was trying on a Windows Docker setup, but did not have luck reaching it and have not had time to dig too far.  Maybe I should consider running the 64 bit Ubuntu though on my Pi and see as a quick test.

**Reply by RUHavingFun** [reply]
So I tried the Ubuntu 64 bit and it worked but seemed sluggish as I had the GUI on.  I then decided to use the 64bit Raspbian (had to fix an issue with a non 64 bit library - https://blog.samcater.com/fix-workaround-rpi4-docker-libseccomp2-docker-20/).  Once all running I am now getting a 4th cam stably on the same overclocked Pi4.  This is good since I have a 2x2 monitor on an Amazon Fire tablet which now stays on all day showing my RTSP feeds.  

I have also found that the viewer I was using on my Mac is buffer prone.  IINA will show it is buffering and that is annoying.  Other viewers (VLC or MVP) just are slower or laggy just a bit but then move right on, which to me is preferred.  So you do not notice the buffering practically at all on those viewers if there is any.

**Reply by RUHavingFun** [reply]
I am now running it on a Raspberry Pi4 2GB overclocked running Raspbian 64bit as it was not using the extra memory at all.

**Comment by dltacube** [comment]
Have you guys tried using RTSP firmware and would you say that this solution is faster that?

**Reply by RUHavingFun** [reply]
Yes.

RTSP firmware is not available for the V3.  I have used and setup RTSP extensively on the V2.  All of my parents V2 cams use that firmware.  

There are a few issues.  Those cameras then are behind or lacking in features as that firmware is rarely updated.  Also, the RTSP feed seems to not be great.  My parents use Blue Iris to record them and continuously have them buffer or time out for long periods.  

Those feeds happen on camera and change the responsiveness of the cameras in general.  This solution is off cam and so fully dependent on the system you install it on.  I have had very reliable feeds once I optimized and found for my setup the max number of cams each Raspberry Pi can handle.  For my setup that is overclocked and handles 3 cams well.

**Comment by RUHavingFun** [comment]
Not sure it anyone else would find interesting.  I made a quick diagram of my setup to explain it to my father for what we might setup for him.![image](https://user-images.githubusercontent.com/39495965/131267469-d1b7873c-1008-4eda-869a-ab262fc86026.png)

[Wyze V3 RTSP Docker Bridge.pdf](https://github.com/mrlt8/docker-wyze-bridge/files/7073321/Wyze.V3.RTSP.Docker.Bridge.pdf)
